### PR TITLE
D8 nid 460 landing page duplicates and redirectds

### PR DESCRIPTION
**Pull request by johangant**
Two key things in this PR:

1. It aims to avoid creating redirects for new landing pages when they already exist.
2. It aims to warn editors when saving a new landing page whether another already exists for that theme value. That should ward off any duplicate content issues and, conversely, also reduce the likelihood of duplicate redirects being created.

There's a fair bit of refactoring to move a lot of code shared between entity insert + update into a separate .inc file for better readability.

### CODE REVIEW

File: nidirect_common/inc/landing_page.redirects.inc [hunk 1]
```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains functions Layout Builder search indexing.
```
Comment (omahm) [inline nidirect_common/inc/landing_page.redirects.inc]: Think you need to update the file description

File: nidirect_common/inc/landing_page.redirects.inc [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ */
+function nidirect_common_get_redirect_url(string $source_path) {
+  $redirect = \Drupal::service('redirect.repository')->findBySourcePath($source_path);
+
+  if (!empty(($redirect))) {
+    $redirect = array_pop($redirect);
+    $redirect_url = $redirect->getRedirectUrl()->toString();
+  }
+  else {
+    $redirect_url = '';
```
Comment (omahm) [inline nidirect_common/inc/landing_page.redirects.inc]: I'd assign this var at the top and then you can do away with the else statement.

File: nidirect_common/inc/landing_page.redirects.inc [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      'language' => Language::LANGCODE_NOT_SPECIFIED,
+      'status_code' => '301',
+    ])->save();
+  }
+  else {
+    // Notify the user that a redirect could not be created.
+    \Drupal::messenger()->addWarning(t('A redirect from the theme taxonomy page to this content was not created,
+        because there are already one or more redirects used for the theme taxonomy page.'));
+
+    // Add a convenience link for users who can administer redirects.
```
Comment (omahm) [inline nidirect_common/inc/landing_page.redirects.inc]: Like it 👍

File: nidirect_common/inc/landing_page.redirects.inc (hunk 1 above)
Comment (johangant) [reply nidirect_common/inc/landing_page.redirects.inc]: *facepalm* My copy/pasta secrets must not be known :)

Good spot!

File: nidirect_common/inc/landing_page.redirects.inc (hunk 2 above)
Comment (johangant) [reply nidirect_common/inc/landing_page.redirects.inc]: Yep, agreed. Like this

**Pull request merged**